lib/vfscore/vfs.h

```diff
@@ -69,89 +69,728 @@ extern int vfs_debug;
 		off_t offset, size_t *count);
-int	 sys_lseek(struct vfscore_file *fp, off_t off, int type, off_t * cur_off);
-int	 sys_ioctl(struct vfscore_file *fp, unsigned long request, void *buf);
-int	 sys_fstat(struct vfscore_file *fp, struct stat *st);
-int	 sys_fstatfs(struct vfscore_file *fp, struct statfs *buf);
-int	 sys_fsync(struct vfscore_file *fp);
-int	 sys_ftruncate(struct vfscore_file *fp, off_t length);
-
-int	 sys_readdir(struct vfscore_file *fp, struct dirent *dirent);
-int	 sys_rewinddir(struct vfscore_file *fp);
-int	 sys_seekdir(struct vfscore_file *fp, long loc);
-int	 sys_telldir(struct vfscore_file *fp, long *loc);
-int	 sys_fchdir(struct vfscore_file *fp, char *path);
-
-int	 sys_mkdir(char *path, mode_t mode);
-int	 sys_rmdir(char *path);
-int	 sys_mknod(char *path, mode_t mode);
-int	 sys_rename(char *src, char *dest);
-int	 sys_link(char *oldpath, char *newpath);
-int	 sys_unlink(char *path);
-int	 sys_symlink(const char *oldpath, const char *newpath);
-int	 sys_access(char *path, int mode);
-int	 sys_stat(char *path, struct stat *st);
-int	 sys_lstat(char *path, struct stat *st);
-int	 sys_statfs(char *path, struct statfs *buf);
-int	 sys_truncate(char *path, off_t length);
-int	 sys_readlink(char *path, char *buf, size_t bufsize, ssize_t *size);
-int  sys_utimes(char *path, const struct timeval *times, int flags);
-int  sys_utimensat(int dirfd, const char *pathname,
+
+/**
+ * Repositions read/write file cursor
+ *
+ * @param fp
+ *	Pointer to the vfscore_file structure
+ * @param off
+ *	Offset bytes to reposition the cursor, according to the type
+ * @param type
+ *	 It can be one of the following types:
+ *		* SEEK_SET: The file offset is set to offset bytes.
+ *		* SEEK_CUR: The file offset is set to its current
+ *			location plus offset bytes.
+ *		* SEEK_END: The file offset is set to the size of
+ *			the file plus offset bytes.
+ * @param[out] cur_off
+ *		* The resulting offset measured from the start of the file
+ * @return
+ *		0 - on success, error code on failure
+ */
+int	sys_lseek(struct vfscore_file *fp, off_t off, int type,
+		off_t *cur_off);
+
+/**
+ * Call for device-specific input/output which cannot be done via other
+ * system calls.
+ *
+ * @param fp
+ *	Pointer to vfscore_file structure
+ * @param request
+ *	Device-dependent request code
+ * @param buf
+ *	Pointer to the data that should be used by call.
+ * @return
+ *	0 - on success, error code on failure
+ */
+int	sys_ioctl(struct vfscore_file *fp, unsigned long request, void *buf);
+
+/**
+ * Syncronizes the in-core data refered by vfscore_file to the disk device
+ *
+ * @param fp
+ *	Pointer to vfscore_file structure
+ * @return
+ *	0 - on success, error code on failure
+ */
+int	sys_fsync(struct vfscore_file *fp);
+
+/**
+ * Gets the next directory entry in directory stream
+ *
+ * @param fp
+ *	Pointer to vfscore_file structure
+ * @param[out] dirent
+ *	Pointer to the directory that will be returned.
```

Comment:
```suggestion
 *	Pointer to the directory that will be returned
```
Only have dots at the end of complete sentences.